CMakeLists.txt

```diff
@@ -91,10 +91,15 @@ endif()
 option(PYBIND11_INSTALL "Install pybind11 header files?" ${PYBIND11_MASTER_PROJECT})
 option(PYBIND11_TEST "Build pybind11 test suite?" ${PYBIND11_MASTER_PROJECT})
 option(PYBIND11_NOPYTHON "Disable search for Python" OFF)
+option(PYBIND11_USE_PYPY_GIL "Force use of PYPY GIL acquire/release logic in gil.h" OFF)
 set(PYBIND11_INTERNALS_VERSION
     ""
     CACHE STRING "Override the ABI version, may be used to enable the unstable ABI.")
 
+if(PYBIND11_USE_PYPY_GIL)
+  add_compile_definitions(USE_PYPY_GIL)
```

Comment:
I'd leave the compile definition scoped (so use the same name).

Reply:
@henryiii We know that the non-simple code is buggy beyond a reasonable doubt: bug reports & @jbms careful analysis. @wjakob himself dropped this code in nanobind and never spoke up for keeping it here, despite getting tagged at least a couple times before. I already ran Google global testing for the simple version, which is the same as in nanobind (except for the missing deletes there) and also what the is used by pretty much everybody else in some form or shape.

Unless someone can gives us new information we don't have here yet, I believe:

* Releasing this code not knowing it is buggy was an accident.

* Releasing this code knowing what we do now is intentionally causing harm. IOW it is irresponsible.

Reply:
How about calling this "PYBIND11_SIMPLE_GIL". It defaults to OFF in 2.10, ON in 2.11. If NanoGUI can be fixed to use "OFF", then it can be removed with a pointer to whatever NanoGUI did as a guide for upgrading. If it can't, then we need to either keep it, or drop it in 3.0.

Reply:
FYI, NanoGUI currently grabs the latest released Pybind11 version from pip, so if this was released without the current impl or even the default changed, NanoGUI would instantly break.

Reply:
```suggestion
  add_compile_definitions(PYBIND11_SIMPLE_GIL)
```